### PR TITLE
Index `Upgrade` API specs and tests for them

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
API specs and tests for upgrade api
And marking them deprecated
GET /_upgrade
POST /_upgrade
GET /{index}/_upgrade
POST /{index}/_upgrade

### Issues Resolved
[#663]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
Let's find out when this was deprecated and why and put a real version number here. What message do you get (aka how do you know it's deprecated)? Look for it in https://github.com/opensearch-project/opensearch and narrow down the commit / PR that did it.

File: tests/default/indices/upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      refresh: true
+    request:
+      content_type: application/x-ndjson
+      payload:
+        - {create: {_index: movies, _id: movie1}}
+        - {director: Bennett Miller, title: The Cruise, year: 1998}
+        - {create: {_index: movies, _id: movie2}}
+        - {director: Nicolas Winding Refn, title: Drive, year: 1960}
+chapters:
+  - synopsis: Check index upgrade status (GET).
```
Comment (dblock) [inline tests/default/indices/upgrade.yaml:21]: Probably good to flip this one with the triggering of an upgrade so we can get actual results and check schema.

**Comment by Tokesh** [comment]
> Let's find out when this was deprecated and why and put a real version number here. What message do you get (aka how do you know it's deprecated)? Look for it in https://github.com/opensearch-project/opensearch and narrow down the commit / PR that did it.

Hello! I found that, the _upgrade API was deprecated in [Elasticsearch 7.10](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/indices-upgrade.html), which is precisely the version OpenSearch [was forked from](https://opensearch.org/faq/#q1.8).

File: tests/default/indices/upgrade.yaml (hunk 1 above)
Comment (Tokesh) [reply tests/default/indices/upgrade.yaml]: If I understood you correctly, then I have corrected

File: tests/default/indices/upgrade.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  - synopsis: Check index upgrade status (GET).
+    path: /{index}/_upgrade
+    method: GET
+    parameters:
+      index: [movies]
+    response:
+      status: 200
+
+  - synopsis: Trigger upgrade (POST).
```
Comment (dblock) [inline tests/default/indices/upgrade.yaml:42]: Note that this is not an `/{index}` API and should go in its own `tests/default/_core/upgrade.yaml`. I'll merge now, but appreciate it being split in a future PR.

**Review by dblock — APPROVED** [review]

**Pull request merged**